### PR TITLE
Support for Bitbucket Server downloading

### PR DESCRIPTION
**Pull request by ametzger**
Adds support for downloading users, projects, repositories, branches, commits, and pull requests for on-premise Bitbucket servers.

### CODE REVIEW

**Comment by mattklein** [comment]
Meta comment, applies to both this and the Jira agent code (@eliwind) -- it's a bit concerning that the jf_agent repo is introducing code that's essentially duplicated in the jellyfish repo. At some point it's going to be painful and error-prone for us to be maintaining both copies of code, there might be subtle differences, we'll forget to fix stuff in both places, etc. What'd be the path to unify them? Make jf_agent a dependency in jellyfish, and refactor the adapter "pull" code to use common code from jf_agent?

File: jf_agent/main.py [hunk 1]
```diff
@@ -2,49 +2,58 @@
...
+    parser.add_argument(
+        '--jira-url',
+        type=str,
+        help='the fully-qualified base URL for your Jira server, e.g. https://jira.yourcompany.com',
+        required=False)
+    parser.add_argument(
+        '--bitbucket-url',
+        type=str,
+        help='the fully-qualified base URL for your Bitbucket server, e.g. https://bitbucket.yourcompany.com',
+        required=False)
```
Comment (eliwind) [inline jf_agent/main.py]: I feel like we should support BB cloud, GH cloud, and GHE too.  Fine to start with this one, but might want to think how the configuration works in general.

Comment (eliwind) [reply jf_agent/main.py]: the rationale for cloud providers being, they may still want to audit what info we're getting, and especially if we have a model where we ever read a local repo clone, it'll have to be agent-based.

File: jf_agent/main.py [hunk 2]
```diff
@@ -2,49 +2,58 @@
...
-    jira_url = args.jira_url
     outdir = args.out_dir
-    
-    if jira_username and jira_password:
-        jira_connection = get_basic_jira_connection(jira_url, jira_username, jira_password)
-    else:
-        print('Jira credentials not found. Set environment variables, either JIRA_SHARED_SECRET, JIRA_CLIENT_KEY, JIRA_KEY (for OAuth) or JIRA_USERNAME and JIRA_PASSWORD (for basic auth)')
+    jira_url = args.jira_url
+    bb_url = args.bitbucket_url
+    skip_ssl_verification = args.no_verify_ssl
```
Comment (eliwind) [inline jf_agent/main.py]: Seems weird to expose this knob for bb but not jira

**Pull request merged**